### PR TITLE
Make _some_ cameras iterable

### PR DESCRIPTION
**Pull request by RealOrangeOne**
For some cameras, the interface to be able to iterate over them is very useful. Some cameras it's not, so it's not implemented.

I mainly noticed this when trying to use the video camera, as there was no easy way to know when you ran out of frames.

### CODE REVIEW

File: zoloto/marker_dict.py [hunk 1]
```diff
@@ -25,3 +26,7 @@ class MarkerDict(IntEnum):
     DICT_APRILTAG_25H9 = aruco.DICT_APRILTAG_25H9
     DICT_APRILTAG_36H10 = aruco.DICT_APRILTAG_36H10
     DICT_APRILTAG_36H11 = aruco.DICT_APRILTAG_36H11
+
+
+def all_marker_dicts() -> Generator[MarkerDict, None, None]:
```
Comment (trickeydan) [inline zoloto/marker_dict.py]: I'm pretty sure that `Enum` is iterable directly?

```python
>>> class Bees(IntEnum):
...     HONEY = 0
...     BUMBLE = 1
...     KILLER = 2
...     CARPENTER = 3
... 
>>> [v for v in Bees]
[<Bees.HONEY: 0>, <Bees.BUMBLE: 1>, <Bees.KILLER: 2>, <Bees.CARPENTER: 3>]
>>> [v.value for v in Bees]
[0, 1, 2, 3]
```

File: zoloto/cameras/camera.py [hunk 2]
```diff
@@ -49,6 +49,13 @@ def discover(cls, **kwargs: Any) -> Generator["Camera", None, None]:
         for camera_id in find_camera_ids():
             yield cls(camera_id, **kwargs)
 
+    def __iter__(self) -> Generator[ndarray, None, None]:
```
Comment (trickeydan) [inline zoloto/cameras/camera.py]: The code for this method is identical between the three `Camera`s that have it.

I'd prefer if you used an `IterableCameraMixin`, or similar.

e.g

```python
class IterableCameraMixin(ABC):

   def __iter__(self) -> Generator[ndarray, None, None]:
        while True:
            frame = self.capture_frame()
            if not frame.size:
                break
            yield frame

class Camera(BaseCamera, IterableCameraMixin):
    ...
```

File: zoloto/marker_dict.py (hunk 1 above)
Comment (RealOrangeOne) [reply zoloto/marker_dict.py]: waaaaat. Not sure how I forgot about this :man_facepalming:

File: zoloto/cameras/camera.py (hunk 2 above)
Comment (RealOrangeOne) [reply zoloto/cameras/camera.py]: I too was thinking this, I forget why I never went for it. Will give it a play

**Pull request merged**